src/Microsoft.Sbom.Api/Workflows/SBOMRedactionWorkflow.cs

```diff
@@ -17,17 +24,133 @@ public class SbomRedactionWorkflow : IWorkflow<SbomRedactionWorkflow>
 
     private readonly IConfiguration configuration;
 
+    private readonly IFileSystemUtils fileSystemUtils;
+
+    private readonly ValidatedSBOMFactory validatedSBOMFactory;
+
+    private readonly SbomRedactor sbomRedactor;
+
     public SbomRedactionWorkflow(
         ILogger log,
-        IConfiguration configuration)
+        IConfiguration configuration,
+        IFileSystemUtils fileSystemUtils,
+        ValidatedSBOMFactory validatedSBOMFactory = null,
+        SbomRedactor sbomRedactor = null)
     {
         this.log = log ?? throw new ArgumentNullException(nameof(log));
         this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
+        this.validatedSBOMFactory = validatedSBOMFactory ?? new ValidatedSBOMFactory();
+        this.sbomRedactor = sbomRedactor ?? new SbomRedactor();
```

Comment:
This sort of pattern to optionally override parameters feels a bit out of place in an app that uses Dependency Injection as a key part of its structure. If we don't want to use DI, would it make sense to split this constructor into 2 constructors, one of which is used for testing?  Something like this:

```suggestion
    // production constructor
    public SbomRedactionWorkflow(
        ILogger log,
        IConfiguration configuration,
        IFileSystemUtils fileSystemUtils)
        : this (log, configuration, fileSystemUtils, new ValidatedSBOMFactory(), new SbomRedactor)
    {
    }
    
    // unit test constructor
    public SbomRedactionWorkflow(
        ILogger log,
        IConfiguration configuration,
        IFileSystemUtils fileSystemUtils,
        ValidatedSBOMFactory validatedSBOMFactory,
        SbomRedactor sbomRedactor)
    {
        this.log = log ?? throw new ArgumentNullException(nameof(log));
        this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
        this.fileSystemUtils = fileSystemUtils ?? throw new ArgumentNullException(nameof(fileSystemUtils));
        this.validatedSBOMFactory = validatedSBOMFactory ?? throw new ArgumentNullException(nameof(validatedSBOMFactory));
        this.sbomRedactor = sbomRedactor ?? throw new ArgumentNullException(nameof(sbomRedactor));
    }
```